src/core/geometry/qgsgeometryeditutils.cpp

```diff
@@ -26,11 +26,12 @@ email                : marco.hugentobler at sourcepole dot com
 #include "qgsvectorlayer.h"
 #include <limits>
 
-int QgsGeometryEditUtils::addRing( QgsAbstractGeometry *geom, QgsCurve *ring )
+QgsGeometry::OperationResult QgsGeometryEditUtils::addRing( QgsAbstractGeometry *geom, QgsCurve *r )
 {
+  std::unique_ptr<QgsCurve> ring( r );
```

Comment:
Is this correct? The way I read the original code is that since "r" is always transferred, it's deleted if it could not be added to the geometry (otherwise there's a leak)